### PR TITLE
Fix AppSec manual blocking parameters

### PR DESCRIPTION
**Pull request by uurien**
### What does this PR do?
Fixes a bug where request blocking through the AppSec SDK wouldn't take into account blocking parameters from the rules file or Remote Config.

### Motivation
Bug report

### CODE REVIEW

File: packages/dd-trace/src/appsec/blocking.js [hunk 1]
```diff
@@ -23,7 +25,7 @@ function addSpecificEndpoint (method, url, type) {
   detectedSpecificEndpoints[getSpecificKey(method, url)] = type
 }
 
-function getBlockWithRedirectData (rootSpan, actionParameters) {
+function getBlockWithRedirectData (actionParameters) {
   let statusCode = actionParameters.status_code
```
[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/src/appsec/blocking.js:29]: meh, not a fan but fine

[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/src/appsec/blocking.js]: no but it doesn't make sense, then what do I put in location header meh

[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/src/appsec/blocking.js:29]: I'd rather it crash when someone one day use it wrong

File: packages/dd-trace/src/appsec/rule_manager.js [hunk 2]
```diff
@@ -19,6 +21,11 @@ function loadRules (config) {
     : require('./recommended.json')
 
   waf.init(defaultRules, config)
+
+  if (defaultRules.actions) {
+    const action = defaultRules.actions.find(action => action.id === 'block')
+    blocking.setDefaultBlockingActionParameters(action?.parameters)
+  }
```
[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/src/appsec/rule_manager.js:28]: Good point, changing it now

[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/src/appsec/rule_manager.js]: Done

File: packages/dd-trace/src/appsec/blocking.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (iunanua) [reply packages/dd-trace/src/appsec/blocking.js:29]: :)

and moving `actionParameters?.location` check inside `getBlockWithRedirectData`?
if there is no `actionParameters?.location` return `undefined`

```javascript
function getBlockingData (req, specificType, actionParameters) {
  return getBlockWithRedirectData(actionParameters) ||
    getBlockWithContentData(req, specificType, actionParameters)
}
```

[the remark this answers is not in the document]
Comment (simon-id) [reply packages/dd-trace/src/appsec/blocking.js:29]: this whole file needs to be rewritten anyway, this is a bugfix PR that i'm trying to merge fast, we can take the time to improve the code later, wdyt ?

[the remark this answers is not in the document]
Comment (iunanua) [reply packages/dd-trace/src/appsec/blocking.js:29]: okok xD

**Review by iunanua — DISMISSED** [review]

**Comment by iunanua** [comment]
ouch a test is failing

**Comment by simon-id** [comment]
yeah it's normal, i'm updating the test after making the change you suggested

**Review by iunanua — APPROVED** [review]

**Pull request merged**